.github/workflows/earlyaccess-unit.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: JDK-EA Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  applications:
+    name: Unit Test EA
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        java-version: [15, 16]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@v2
+      - name: 'Download JDK'
+        uses: sormuras/download-jdk@v1
+        id: download-jdk
+        with:
+          feature: ${{ matrix.java-version }}
+      - name: 'Set up JDK'
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ steps.download-jdk.outputs.version }}
+          jdkFile: "${{ steps.download-jdk.outputs.file }}"
+      - name: Set Maven Wrapper
+        run: mvn -N io.takari:maven:0.7.7:wrapper -Dmaven=3.6.3
+      - name: 'JDK Version'
+        run: java --version
+      - name: Unit-Test
+        run: ./mvnw verify --no-transfer-progress
```

Comment:
create a new profile in the top level pom.xml that excludes the p2 stuff (see this [example](https://stackoverflow.com/questions/4448599/maven-modules-optional-in-pom-xml)). Call it `ea-profile`. Then specify it on this line with `-P ea-profile`.

Reply:
We need to test the p2 stuff as well. Also the profile part is not working. If you look at earlyaccess-javadoc.yml in this PR, I specifically give the modules to build and still p2 repo is getting built. That is the whole deal with EBR plugin and I do not know how to solve that. I am working on fixing the problem with EBR plugin itself. PR raised there: https://github.com/eclipse/ebr/pull/40